### DISCUSSION TITLE
system.file.allocate recommandation

### DISCUSSION THREAD

**Discussion by simonc56** [crazy-max/docker-rtorrent-rutorrent#387] [Q&A] [answered]
I use rtorrent @stickz BitTorrent client version 0.9.8-v6.0 on an arm SBC with SSD storage.

Is it recommanded to use :
```
system.file.allocate.set=1
``` 
or not in _.rtorrent.rc_ config file ?

I see in the docker container the file _/etc/rtorrent/.rtlocal.rc_ having :
```
# Configure preallocation feature for rTorrent client
system.file.allocate.set = 0
```

In 2000s-2010s when I used old style HDD I thought enabling allocate was a better practice to prevent fragmentation.
But in 2024 what is the best option ?

**Comment by Undrog** [comment, marked as answer]
There's a short section about this setting in the main README.md for this project. It's disabled by default because it only benefits HDD's according to this. From what I know about SSD's, I'd agree that preallocation doesn't do any good with SSD, but if you are using HDD's, then you might want to consider it to reduce fragmentation/enable fast sequential reads without significant seeking.

**Reply by simonc56** [reply]
thanks, i'll disable it.
